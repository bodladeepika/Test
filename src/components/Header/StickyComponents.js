import React,{Fragment,useState} from 'react'

export default () => {
    const [isOpen, setOpen] = useState(false);
    const dropdownClass = `dropdown-menu${isOpen ? " show" : ""}`;

    const toggleOpen = () => {
        setOpen(!isOpen);
    }
   
    return (
        <Fragment> 
           
            <div className="sticky-inner">
           
            <div class="dropdown mr-4" onClick={toggleOpen}>
                <a class="btn btn-secondary dropdown-toggle" href="#" role="button" id="dropdownMenuLink" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
                    
                </a>

                <div class={dropdownClass} aria-labelledby="dropdownMenuLink">
                    <a class="dropdown-item" href="#">Acount Settings</a>
                    <a class="dropdown-item" href="#">User Management</a>
                    <a class="dropdown-item" href="#">My Team</a>
                    <a class="dropdown-item" href="#">English(UK)</a>
                    <a class="dropdown-item" href="#">Logout</a>
                </div>
                </div>

            </div>
        </Fragment>
    )
}