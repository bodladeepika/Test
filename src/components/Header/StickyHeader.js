import React, { Fragment, useEffect, useRef, useState } from 'react';
import StickyComponents from './StickyComponents';
import './header.scss'


export default () => {
  const [isSticky, setSticky] = useState(false);
  const ref = useRef(null);
  const handleScroll = () => {
    if (ref.current) {
      
      setSticky(ref.current.getBoundingClientRect().top <= 0);
    }
  };

  useEffect(() => {
    window.addEventListener('scroll', handleScroll);

    return () => {
      window.removeEventListener('scroll', () => handleScroll);
    };
  }, []);

  return (
    <Fragment>
    
      <div className="">
           <div id="header"> 
            <div className={`sticky-wrapper${isSticky ? ' sticky' : ''}`} ref={ref}>
              <StickyComponents/>
            </div>
            </div>
        
        
      </div>
    
      
    </Fragment>
  );
};