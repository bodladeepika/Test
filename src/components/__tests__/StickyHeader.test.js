import React from 'react';
import StickyHeader from '../Header/StickyHeader'
import {shallow} from 'enzyme'
import StickyComponents from '../Header/StickyComponents';
import toJson from 'enzyme-to-json';
import '@testing-library/jest-dom/extend-expect';


let wrapped;
beforeEach ( () => {
    wrapped = shallow(<StickyHeader/>);
})

afterEach ( () => {
    wrapped.unmount()
})




it('has another component' , () => {

expect(wrapped.containsMatchingElement(<StickyComponents />)).toEqual(true);
})

it('has correclty rendered',()=>{
  expect(toJson(wrapped)).toMatchSnapshot()
})

