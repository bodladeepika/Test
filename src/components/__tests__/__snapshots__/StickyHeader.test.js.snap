// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has correclty rendered 1`] = `
<Fragment>
  <div
    className=""
  >
    <div
      id="header"
    >
      <div
        className="sticky-wrapper"
      >
        <_default />
      </div>
    </div>
  </div>
</Fragment>
`;
